frontend/public/components/pod.tsx

```diff
@@ -484,6 +489,67 @@ export const ContainerLink: React.FC<ContainerLinkProps> = ({ pod, name }) => (
 );
 ContainerLink.displayName = 'ContainerLink';
 
+const ContainerRunningSince: React.FC<ContainerRunningSinceProps> = ({ startedAt }) => {
+  const { t } = useTranslation();
+  return startedAt ? (
+    <Trans t={t} ns="public">
+      since <Timestamp timestamp={startedAt} simple />
+    </Trans>
+  ) : null;
+};
+
+const ContainerTerminatedAt: React.FC<ContainerTerminatedAtProps> = ({ finishedAt }) => {
+  const { t } = useTranslation();
+  return finishedAt ? (
+    <Trans t={t} ns="public">
+      at <Timestamp timestamp={finishedAt} simple />{' '}
+    </Trans>
+  ) : null;
+};
+
+const ContainerTerminatedExitCode: React.FC<ContainerTerminatedExitCodeProps> = ({ exitCode }) => {
+  const { t } = useTranslation();
+  return exitCode ? t('public~with exit code {{exitCode}} ', { exitCode }) : null;
+};
+
+const ContainerTerminatedReason: React.FC<ContainerTerminatedReasonProps> = ({ reason }) => {
+  const { t } = useTranslation();
+  return reason ? t('public~({{reason}})', { reason }) : null;
+};
+
+export const ContainerLastState: React.FC<ContainerLastStateProps> = ({ containerLastState }) => {
+  const { t } = useTranslation();
+  if (containerLastState?.waiting) {
+    return t('public~Waiting {{reason}}', { reason: containerLastState.waiting?.reason });
+  } else if (containerLastState?.running) {
+    return (
+      <Trans t={t} ns="public">
+        Running <ContainerRunningSince startedAt={containerLastState.running?.startedAt} />
+      </Trans>
+    );
+  } else if (containerLastState?.terminated) {
+    return (
+      <Trans t={t} ns="public">
+        Terminated <ContainerTerminatedAt finishedAt={containerLastState.terminated?.finishedAt} />
```

Comment:
I would rather have the render logic be part of a single component rather then three separate ones, since they are not used anywhere.

Reply:
The separate components are necessary since the data is optional